app/src/main/java/org/fossasia/openevent/general/event/EventDetailsViewModel.kt

```diff
@@ -5,16 +5,32 @@ import android.arch.lifecycle.ViewModel
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.disposables.CompositeDisposable
 import io.reactivex.schedulers.Schedulers
+import org.fossasia.openevent.general.auth.DiscountCode
+import org.fossasia.openevent.general.discount.DiscountService
 import timber.log.Timber
 
-class EventDetailsViewModel(private val eventService: EventService) : ViewModel() {
+class EventDetailsViewModel(private val eventService: EventService, private val discountService: DiscountService) : ViewModel() {
 
     private val compositeDisposable = CompositeDisposable()
 
     val progress = MutableLiveData<Boolean>()
     val event = MutableLiveData<Event>()
+    val discontCode = MutableLiveData<DiscountCode>()
     val error = MutableLiveData<String>()
 
+    fun loadDiscountCode(id : Long) {
+        compositeDisposable.add(discountService.getDiscountCode(id)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doOnError({
+                    discountService.getDiscountCode(id)
+                })
+                .doOnSuccess({
+                    discontCode.value = it
+                })
```

Comment:
These are side effects, never use them. Do this in subscribe

If there's an error, the app will crash throwing OnErrorNotImplemented